### PR TITLE
Add support for merge-includes in worlds

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
Implements merge-includes for worlds as discussed in https://github.com/gazebosim/sdf_tutorials/pull/86. This required very few changes since it uses the same merge mechanism used for models. 

## Test it
Two tests were added in `test/integration/includes.cc`. 

`IncludeTest.WorldMergeInclude`: Checks the proper merging of elements and their poses. 
`IncludeTest.WorldMergeIncludeInvalidElements. Checks that merging models that contain links into the world emits an error.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- ~[ ] Consider updating Python bindings (if the library has them)~
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: Migration.md
Comment (aaronchongth) [inline Migration.md:561]: From https://github.com/azeey/sdformat/blob/world_merge_include/src/parser.cc#L374, it looks like `gripper` is not supported as well.

File: test/integration/includes.cc
Comment (aaronchongth) [inline test/integration/includes.cc:573]: nit: extra line here

**Review by aaronchongth — CHANGES_REQUESTED** [review]
This is awesome! I only have 2 small comments, but otherwise LGTM!

File: Migration.md
Comment (azeey) [reply Migration.md]: Done in https://github.com/gazebosim/sdformat/pull/1233/commits/e1216672856481b1cfcce590546e2d1d8c676e8e

File: test/integration/includes.cc
Comment (azeey) [reply test/integration/includes.cc:573]: https://github.com/gazebosim/sdformat/pull/1233/commits/e1216672856481b1cfcce590546e2d1d8c676e8e

**Review by aaronchongth — APPROVED** [review]
LGTM!

File: Migration.md
Comment (EricCousineau-TRI) [inline Migration.md:561]: nit Consider qualifying:

"The included models must not contain **top-level** links or grippers."

File: src/parser.cc
Comment (scpeters) [inline src/parser.cc:378]: weird that coverage doesn't detect this because I see that it is tested

**Review by scpeters — APPROVED** [review]

File: src/parser.cc
Comment (azeey) [reply src/parser.cc:378]: I think coverage struggles with multiline code.

File: Migration.md
Comment (azeey) [reply Migration.md]: Done in https://github.com/gazebosim/sdformat/pull/1233/commits/f6a975222f4f6688600be7f364c09ec497d97d62

Comment (EricCousineau-TRI) [reply Migration.md]: Thanks!

**Review by EricCousineau-TRI — APPROVED** [review]

**Comment by scpeters** [comment]
I think the homebrew CI errors will be fixed by the gz-math7 bottle rebuild in https://github.com/osrf/homebrew-simulation/pull/2213

**Comment by scpeters** [comment]
> I think the homebrew CI errors will be fixed by the gz-math7 bottle rebuild in [osrf/homebrew-simulation#2213](https://github.com/osrf/homebrew-simulation/pull/2213)

rerunning homebrew CI since osrf/homebrew-simulation#2213 has been merged

**Comment by scpeters** [comment]
green CI! I just merged with upstream and will leave it to @azeey to merge

**Pull request merged**
